airflow/www/static/js/dag/details/gantt/index.tsx

```diff
@@ -136,21 +134,30 @@ const Gantt = ({ openGroupIds, gridScrollRef, ganttScrollRef }: Props) => {
 
       if (end && (!endDate || Date.parse(end) > Date.parse(endDate))) {
         setEndDate(end);
+      } else if (!end) {
+        // @ts-ignore
+        setEndDate(moment().add(1, "s").toString());
       }
     },
     [startDate, endDate, setStartDate, setEndDate]
   );
 
+  // Reset state when the dagrun changes
   useEffect(() => {
-    groups.children?.forEach((task) => {
```

Comment:
Instead of iterating through the array multiple times, let's just call setGanttDuration from each row component.